stubs/PyMySQL/pymysql/converters.pyi

```diff
@@ -1,47 +1,39 @@
-ESCAPE_REGEX: Any
-ESCAPE_MAP: Any
-
-def escape_item(val, charset): ...
-def escape_dict(val, charset): ...
-def escape_sequence(val, charset): ...
-def escape_set(val, charset): ...
-def escape_bool(value): ...
-def escape_object(value): ...
-
-escape_int: Any
-
-escape_long: Any
-
-def escape_float(value): ...
-def escape_string(value): ...
-def escape_unicode(value): ...
-def escape_None(value): ...
-def escape_timedelta(obj): ...
-def escape_time(obj): ...
-def escape_datetime(obj): ...
-def escape_date(obj): ...
-def escape_struct_time(obj): ...
-def convert_datetime(connection, field, obj): ...
-def convert_timedelta(connection, field, obj): ...
-def convert_time(connection, field, obj): ...
-def convert_date(connection, field, obj): ...
-def convert_mysql_timestamp(connection, field, timestamp): ...
-def convert_set(s): ...
-def convert_bit(connection, field, b): ...
-def convert_characters(connection, field, data): ...
-def convert_int(connection, field, data): ...
-def convert_long(connection, field, data): ...
-def convert_float(connection, field, data): ...
-
-encoders: Any
-decoders: Any
-conversions: Any
-
-def convert_decimal(connection, field, data): ...
-def escape_decimal(obj): ...
+import datetime
+import time
+from collections.abc import Callable, Mapping, Sequence
+from decimal import Decimal
+from typing import Any, Optional, Type, TypeVar
+
+_EscaperMapping = Optional[Mapping[Type[object], Callable[..., str]]]
+_T = TypeVar("_T")
+
+def escape_item(val: object, charset: object, mapping: _EscaperMapping = ...) -> str: ...
+def escape_dict(val: Mapping[str, object], charset: object, mapping: _EscaperMapping = ...) -> dict[str, str]: ...
+def escape_sequence(val: Sequence[object], charset: object, mapping: _EscaperMapping = ...) -> str: ...
+def escape_set(val: set[object], charset: object, mapping: _EscaperMapping = ...) -> str: ...
+def escape_bool(value: bool, mapping: _EscaperMapping = ...) -> str: ...
+def escape_int(value: int, mapping: _EscaperMapping = ...) -> str: ...
+def escape_float(value: float, mapping: _EscaperMapping = ...) -> str: ...
+def escape_string(value: str, mapping: _EscaperMapping = ...) -> str: ...
+def escape_bytes_prefixed(value: bytes, mapping: _EscaperMapping = ...) -> str: ...
+def escape_bytes(value: bytes, mapping: _EscaperMapping = ...) -> str: ...
+def escape_str(value: str, mapping: _EscaperMapping = ...) -> str: ...
+def escape_None(value: None, mapping: _EscaperMapping = ...) -> str: ...
+def escape_timedelta(obj: datetime.timedelta, mapping: _EscaperMapping = ...) -> str: ...
+def escape_time(obj: datetime.time, mapping: _EscaperMapping = ...) -> str: ...
+def escape_datetime(obj: datetime.datetime, mapping: _EscaperMapping = ...) -> str: ...
+def escape_date(obj: datetime.date, mapping: _EscaperMapping = ...) -> str: ...
+def escape_struct_time(obj: time.struct_time, mapping: _EscaperMapping = ...) -> str: ...
+def Decimal2Literal(o: Decimal, d: _EscaperMapping) -> str: ...
```

Comment:
This should just copy the definition of `builtins.format()` (minus the default):

```python
def format(__value: object, __format_spec: str = ...) -> str: ...
```

Reply:
It actually ignores the second argument. I'll type it as `object`.

And the first argument can only be an object that accepts `"f"` as a format specifier. I don't know what other types that would include, but clearly it's meant for Decimal, so I'd prefer to keep that as the annotation.